### DISCUSSION TITLE
Shutterbuttons with #define USE_SHUTTER_ESP32

### DISCUSSION THREAD

**Discussion by DaBaBW** [arendst/Tasmota#19133] [Q&A]
### Discussed in https://github.com/arendst/Tasmota/discussions/19131

<div type='discussions-op-text'>

<sup>Originally posted by **DaBaBW** July 17, 2023</sup>
Hello,

if I compile including #define USE_SHUTTER_ESP32 I´m not able to define ShutterButtons for more than one shutter.

Shutter 1:
`11:27:43.186 CMD: backlog ShutterButton1 1 up 1; ShutterButton1 2 down 1
11:27:43.204 SHT: About to save settings to file /.drvset027
11:27:43.269 RSL: RESULT = {"ShutterButton1":"1 100 1 75 100 - - - -"}
11:27:43.431 SHT: About to save settings to file /.drvset027
11:27:43.495 RSL: RESULT = {"ShutterButton1":"1 100 1 75 100 - - - -|2 0 1 25 0 - - - -"}`

Shutter 2:
`11:29:12.285 CMD: backlog ShutterButton2 3 up 1; ShutterButton2 4 down 1
11:29:12.304 SHT: About to save settings to file /.drvset027
11:29:12.370 RSL: RESULT = {"ShutterButton2":"-"}
11:29:12.532 SHT: About to save settings to file /.drvset027
11:29:12.599 RSL: RESULT = {"ShutterButton2":"-"}`

Is this a bug in the enhanced ESP32 functions?
</div>

**Comment by Jason2866** [comment]
Moving to discussions, without not filling out the issue template, it is not accepted as valid issue.

**Reply by DaBaBW** [reply]
I first created a discussion, then I clicked the "Create issue from discussion" so there was no template to fill out. (?)

**Comment by Jason2866** [comment]
Suggesting you are using latest development version, there is no compile option `#define USE_SHUTTER_ESP32` anymore it is `#define USE_SHUTTER` for esp8266 and esp32.

**Reply by DaBaBW** [reply]
Really? If I compile without `#define USE_SHUTTER_ESP32` and simply check the "Shutters and Blinds" in online compiler I´m not able to configure more than 4 shutter buttons. (But ShutterButton2 is working then). So I think this is ESP8266 version. Since which version is this changing? [https://tasmota.github.io/docs/Blinds-and-Shutters/](url) tells that I still need the  `#define USE_SHUTTER_ESP32`

**Comment by Jason2866** [comment]
I dont know which version you are using. You dont tell any info about. Quoting my post
>Suggesting you are using latest development version

it is like i described

**Comment by DaBaBW** [comment]
Oh sorry, I forgot. It is the same in all Versions. I tried from V12 to V13.

**Reply by barbudor** [reply]
By making the effort to look at the change log, you would have seen that this ESP32 mode was introduced in [V12.4.0.5](https://github.com/arendst/Tasmota/blob/development/CHANGELOG.md#12405-20230417)

So no need to go back to Mathusalem
As some fix were done in the last week, like Jason advised, stick to development version. 

Also you haven't provided any information about your configuration including but not limited to
- device
- template and gpios configuration

**Comment by DaBaBW** [comment]
I just wanted to know if it is a known bug. So now I will collect all infos and provide them,  soon.

**Comment by DaBaBW** [comment]
Ok, I really don´t know why. I exactly did the same as the last days again and now it is working.
Sorry to bother you!

**Comment by DaBaBW** [comment]
Ok, I was too fast with my answer. It is not working. I only tested Shutter2 the last days.
Now I saw that I´m still not able to configure more than 4 buttons. And configuration isn´t saved to file, too.
So I think the ESP32 functions of shutters are not active when compiled with `#define USE_SHUTTER` even in Tasmota 13.0.0.
Tested on Nodemcu 38pin ESP-32S and one with WROOM-32D with online compiler like the pictures below.

If I compile with `#define USE_SHUTTER_ESP32` configuration is saved to file and I´m able to configure all 8 buttons of PCF8574 but only for Shutter1. Shutter 2 to 4 everytime reply `RSL: RESULT = {"ShutterButton4":"-"}`

![Screenshot 2023-07-21 184243](https://github.com/arendst/Tasmota/assets/36332774/2e77c56b-0051-4826-83f0-5293dd469e39)
![Screenshot 2023-07-21 184428](https://github.com/arendst/Tasmota/assets/36332774/69e9582f-d3e3-4ad9-a6c4-3af8fb250d27)
![Screenshot 2023-07-21 184851](https://github.com/arendst/Tasmota/assets/36332774/6ff1bc14-215b-468b-9b0b-2edf5d0ed333)
![ESP32 Konfig 4-fach Shutter](https://github.com/arendst/Tasmota/assets/36332774/963acde8-dc02-4cf5-93df-3bad18be25d8)
